### ISSUE TITLE
`indent`: wrapped `?:` of a `TSConditionalType` indents against the test-clause column instead of the parent indent

### ISSUE THREAD

**Issue by lo1tuma** [eslint-stylistic/eslint-stylistic#1216]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

For a wrapped `?:` inside a `TSConditionalType`, the `:` continuation is indented relative to the *visual column* of the test clause instead of the logical indent of the line containing `?`. When the conditional shares a line with siblings to its left, their widths leak into the continuation indent and add spurious nesting.

Minimal repro with `'@stylistic/indent': ['error', 4, { VariableDeclarator: 1, MemberExpression: 1 }]`:

```ts
export type Helper<T> =
    | (T extends string ? readonly T[]
        : T extends number ? readonly T[]
        : T);
```

The rule reports:

> Expected indentation of 12 spaces but found 8.

By logical level:

- level 0: the type-alias statement
- level 1 (+4): the `|` operand (continuation of `=`)
- level 2 (+4): the `?:` continuation
- → `:` should be at column 9 (8 spaces)

The rule arrives at 12 because the test clause `T extends string` happens to start at column 9 (it shares a line with `| (`), and the rule treats `(test-clause column) + indent` as the target. That conflates the test clause's visual position with the conditional's nesting depth. The parens are grouping and do not introduce a level. The `|` does not introduce a level for its operand. Only the `?:` introduces a level for its continuation.

The same bug reproduces without parens or unions — any time the conditional shares a line with constructs to its left whose width pushes the test clause's column past `parent-indent + indent`, the `:` indent inherits the extra. With a long object key on the same line as a ternary, the `:` indent grows with the key length, which clearly can't be right.

### Expected behavior

A wrapped `?:` continuation should indent to `indent-of-the-line-containing-? + one indent step`, independent of where the test clause starts on that line. For the repro above, that's `4 + 4 = 8` spaces. This matches Prettier and dprint, and matches how the rule already handles other continuations (e.g. binary-operator continuations are computed against the parent statement's indent, not against the left-operand column).

### Practical impact

TS projects using Prettier or dprint have to disable `@stylistic/indent` on `.ts` files, because unions of conditional types — a very common shape, fight the formatter.

### Related

- #676 — `indent-binary-ops` mistreating TS unions; commenters there also report circular fixes between `@stylistic/indent` and `@stylistic/indent-binary-ops` on conditional-type + union shapes.
- #1075 — recent refactor of `TSConditionalType` handling (behaviour-preserving, didn't touch the column logic).

### Reproduction

Inline code repro above; reproduces in any project with the listed `@stylistic/indent` config. Happy to provide a Stackblitz if needed.

### System Info

```shell
Reproduces against current `main` (`@stylistic/eslint-plugin@6.0.0-beta.4`) and on the latest 5.x.
```

### Contributes

- [x] I am willing to submit a PR to fix this issue
- [x] I am willing to submit a PR with failing tests

**Comment by 9romise** [comment]
Hey, thanks for the detailed report and sorry for the delayed reply! Have you tried the `flatTernaryExpressions` option? It addresses exactly this style of ternary/conditional type formatting:

```json
{
  "@stylistic/indent": ["error", 4, {
    "flatTernaryExpressions": true,
    "VariableDeclarator": 1,
    "MemberExpression": 1
  }]
}
```

With this option enabled, when the test and consequent are on the same line, the indent rule skips checking the `:` continuation, allowing the flat style you're expecting:

```typescript
export type Helper<T> =
    | (T extends string ? readonly T[]
        : T extends number ? readonly T[]
        : T);
```

Does this work for your use case?

**Comment by lo1tuma** [comment]
Thanks @9romise! I did try `flatTernaryExpressions`, but it doesn't produce the expected output. With the exact config from your comment it still reports errors on the example, and `--fix` lands on a different indent than expected. Measured against current `main`:

**`flatTernaryExpressions: false` (default)** reports `Expected indentation of 12 spaces but found 8.` on line 4 only, i.e. it wants the chained `:` one step deeper than the first:

```ts
export type Helper<T> =
    | (T extends string ? readonly T[]
        : T extends number ? readonly T[]
            : T); // forced to 12
```

**`flatTernaryExpressions: true`** still reports errors, but in the opposite direction:

```
Expected indentation of 4 spaces but found 8.  // line 3
Expected indentation of 4 spaces but found 8.  // line 4
```

and `--fix` collapses both `:` to align with the `|`:

```ts
export type Helper<T> =
    | (T extends string ? readonly T[]
    : T extends number ? readonly T[]
    : T); // both : forced to 4
```

So the option doesn't skip the `:` check here. The colons fall back to the enclosing offset (the parenthesized type at the union level), which forces them to 4 and rejects the 8-space form from the issue.

**Expected** (what dprint produces, and what the issue asks for) is one step under the `|`, with the whole chain flat at that level:

```ts
export type Helper<T> =
    | (T extends string ? readonly T[]
        : T extends number ? readonly T[]
        : T); // both : at 8
```

The difference is the reference point. `flatTernaryExpressions` aligns the whole chain to the conditional's enclosing context, whereas the wanted behavior aligns it to `indent-of-the-line-where-the-conditional-starts + one step`, independent of how deep the chain nests.

#1225 fixes exactly this for chained conditional types written compactly (`test ? consequent` on one line) and leaves `flatTernaryExpressions` untouched. The fully broken-out form (a line break after every `?`, as Prettier emits) keeps its existing per-level indentation.
